lisp/ess-inferior-r.el

```diff
@@ -0,0 +1,250 @@
+;;; ess-inferior-r.el --- inferior-r-process         -*- lexical-binding: t; -*-
+
+;; Copyright (C) 2018 ESS Core team
+;;
+;; This file is part of ESS (Emacs Speaks Statistics).
+;;
+;; ESS is free software; you can redistribute it and/or modify it
+;; under the terms of the GNU General Public License as published by
+;; the Free Software Foundation; either version 2, or (at your option)
+;; any later version.
+;;
+;; This file is distributed in the hope that it will be useful, but
+;; WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+;; General Public License for more details.
+;;
+;; A copy of the GNU General Public License is available at
+;; http://www.r-project.org/Licenses/
+
+;;; Commentary:
+;; Support for starting, running, and otherwise interacting with R
+;; inside Emacs buffers. See the ESS Info manual for more information.
+
+;;; Code:
+
+(require 'ess-help)
+(require 'ess-r-mode)
+(require 'ess-inf)
+
+(defvar ess-inferior-r-mode-map
```

Comment:
This is not a conventional namespace. If we're going to create a new ns, it should probably be `inf-ess-` to follow convention. There are several cases where the namespace is not a prefix, e.g. you don't write `ess-with-` but `with-ess-`.

So we should use `inf-` like in `inf-haskell-mode`, `inf-clojure-`, . It's also fine to keep `inferior-ess-` as in `inferior-python-`, `inferior-tcl-`, `inferior-octave-`, though I like the conciseness of `inf`.

One interesting case is python where they define `inferior-python-mode`, with accompanying map and hook variables, but otherwise namespace most functions and variables with `python-shell-`. I would also consider this solution.

Reply:
I'm not a huge fan of abbreviating long words to shorter words, so how about `inferior-ess-r-mode`?

Reply:
ok let's do that, it has the clear advantage of being the status quo. We already have functions in that namespace, e.g. `inferior-ess-r-force`.